### ISSUE TITLE
type_routing plugin does not appear to work for sub apps when loaded on parent app

### ISSUE THREAD

**Issue by kenaniah** [jeremyevans/roda#82]
Given the following app:

```ruby
class App < Roda
  plugin :type_routing, use_headers: false
end

class Sub < App
  route do |r|
    r.get 'a' do
      r.html{ "<h1>This is the HTML response</h1>" }
      r.json{ '{"json": "ok"}' }
      r.xml{ "<root>This is the XML response</root>" }
      "Unsupported data type"
    end
  end
end

class Routes < App
  route do |r|
    r.on "subpath", proc { r.run Sub }
  end
end

class App
  route do |r|
    r.run Routes
  end
end
```

The app does not route by type as expected:

  * `/subpath/a` -> html (works)
  * `/subpath/a.html` -> html (works)
  * `/subpath/a.json` -> html (broken)
  * `/subpath/a.xml` -> html (broken)

**Comment by jeremyevans** [comment]
Thanks for the report! This is because the extension is removed from the remaining_path when the App instance is initialized.  It should add the extension back if run is called.  I'll push a fix shortly.

**Comment by jeremyevans** [comment]
Fixed by f75944b09f1843ff1582533951a20ca75ac06786
